elastic/kibana
# [Remote clusters] Add new security model

Fixes: https://github.com/elastic/kibana/issues/154312

## Summary
This PR takes care of updating the UX for the creation of remote clusters so that after creating a new configuration we show the users the correspondant docs on how to setup authentication for it. Auth can be configured in two ways: either server TLS or mutual TLS, based on what the user selects we need to explain to them how to configure it.


##### Notes
With this PR I also fixed the EUI deprecations from the List,Edit and Add pages.

##### How to test
1. Start elasticsearch with `yarn es snapshot --license=trial` and serverless search solution with `yarn start`
2. Navigate to the remote clusters UI and create a new cluster config
3. Verify that adding a new config has the two steps wizard
4. Verify that the config is created after the second step is compleated
5. Verify that if the user is on at least enterprise license the API Keys method is shown, but if the user is on basic license only certificates should be shown.
6. Verify that if an error happen when creating the config, the user is sent to the first step of the wizard


<details>
<summary>Screenshots empty states</summary>

![Screenshot 2023-07-19 at 08 42 57](https://github.com/elastic/kibana/assets/1191206/8ba59f3a-95de-48f1-a5b8-b5fa6df2b85a)
![Screenshot 2023-07-19 at 08 43 32](https://github.com/elastic/kibana/assets/1191206/2e79e0b7-14e6-49f8-ad76-49e55f381c82)
![Screenshot 2023-07-19 at 08 43 54](https://github.com/elastic/kibana/assets/1191206/6a9c9147-dd7c-48d4-a963-85357d4faec1)
<img width="1504" alt="Screenshot 2023-08-01 at 17 03 43" src="https://github.com/elastic/kibana/assets/1191206/80d05413-c0b0-4670-b781-f0384cc56bb1">


</details>

<details>
<summary>Screenshots </summary>

// Adding configuration flow
![Screenshot 2023-07-19 at 08 44 07](https://github.com/elastic/kibana/assets/1191206/a91a5f35-138b-45c8-9d5b-d7b53599bcd7)
// With local form errors
![Screenshot 2023-07-19 at 09 31 28](https://github.com/elastic/kibana/assets/1191206/db95236b-f641-416c-80b9-94e35d02bf4b)
// With enterprise license
<img width="1506" alt="Screenshot 2023-08-01 at 17 01 12" src="https://github.com/elastic/kibana/assets/1191206/7f4d054e-e24b-471a-bad8-4759ea537c18">
// Without enterprise license
<img width="1495" alt="Screenshot 2023-08-01 at 17 02 03" src="https://github.com/elastic/kibana/assets/1191206/3cc5fe03-4027-40c4-934d-8798071c94cc">
// With API error
![Screenshot 2023-07-20 at 09 04 15](https://github.com/elastic/kibana/assets/1191206/999d03b5-fd03-460f-8de5-b2c7ba3c6c46)

<img width="1497" alt="Screenshot 2023-08-01 at 17 01 37" src="https://github.com/elastic/kibana/assets/1191206/325be12d-1a18-48f2-943e-b8b4d45c9fd6">


![Screenshot 2023-07-19 at 08 44 31](https://github.com/elastic/kibana/assets/1191206/d6bb5d2c-a4a6-4f99-9829-78f41fff76b9)

// Remote clusters list
![Screenshot 2023-07-19 at 08 44 45](https://github.com/elastic/kibana/assets/1191206/8bc2f4dd-ffeb-4570-9b86-2f597c4c90eb)

// Edit remote cluster config
![Screenshot 2023-07-19 at 08 44 51](https://github.com/elastic/kibana/assets/1191206/73bb5567-349d-46d6-b680-785cae64aeab)

</details>
